addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/__test__/launch-product.test.js

```diff
@@ -13,106 +13,105 @@
-            expect(namespaceParam).toBe(`analysis-${originalNamespace}-${datetime}`);
-        });
-
-        it('Static name that ends with a datetime string should be transformed to start with analysis-', async () => {
-            // Build
-            const datetime = Date.now();
-            const resolvedInputParams = [{Key: 'Namespace', Value: `staticname-${datetime}`}];
-            const originalNamespace = resolvedInputParams[0].Value;
-    
-            // Operate
-            const { namespaceParam, namespaceIndex } = lp.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
-    
-            // Check
-            expect(namespaceParam).toBe(`analysis-${originalNamespace}`);
-        });
-
-        it('getNamespaceAndIndexIfNecessary does not change the original resolvedInputParams array when the namespace is static', async () => {
-            // Build
-            const datetime = Date.now();
-            const resolvedInputParams = [{Key: 'Namespace', Value: 'staticname'}];
-
-            // Operate
-            const { namespaceParam, namespaceIndex } = lp.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
-
-            // Check
-            expect(namespaceParam).not.toBe(resolvedInputParams[namespaceIndex]);
-        })
-    })
-})
+describe('LaunchProduct', () => {
+  const meta = {};
+  const input = {};
+
+  const lp = new LaunchProduct({
+    step: { config: {} },
+    workflowPayload: new WorkflowPayload({ meta, input, workflowInstance: { steps: [] } }),
+  });
+
+  describe('checkNamespace', () => {
+    it('Static name should be transformed to start with analysis- and end with number string and be unique between calls', async () => {
+      // Build
+      const resolvedInputParams = [{ Key: 'Namespace', Value: 'staticname' }];
+      const datetime = Date.now();
+      const resolvedInputParams2 = [{ Key: 'Namespace', Value: 'staticname' }];
+      const datetime2 = Date.now() + 1;
+      const originalNamespace = resolvedInputParams[0].Value;
+
+      // Operate
+      const after = lp.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
+      const after2 = lp.getNamespaceAndIndexIfNecessary(resolvedInputParams2, datetime2);
+
+      // Check
+      expect(after.namespaceParam).not.toBe(after2.namespaceParam);
+      expect(after.namespaceParam).toBe(`analysis-${originalNamespace}-${datetime}`);
+    });
+
+    it('Static name that begins with analysis- should be transformed to end with number string and be unique between calls', async () => {
+      // Build
+      const resolvedInputParams = [{ Key: 'Namespace', Value: 'analysis-staticname' }];
+      const resolvedInputParams2 = [{ Key: 'Namespace', Value: 'analysis-staticname' }];
+      const datetime = Date.now();
+      const datetime2 = Date.now() + 1;
+      const originalNamespace = resolvedInputParams[0].Value;
+
+      // Operate
+      const after = lp.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
```

Comment:
Let's make these more readable. What are they supposed to denote? for eg. `updatedParams`

Reply:
Yeah, updatedParams makes more sense. Will change now!